File: CS/PrintCrosshairSample/MainViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace PrintCrosshairSample {
    public class MainViewModel {
        private SaleItemRepository repository;

        public IReadOnlyList<SaleItemAggregate> TotalIncome {
            get => repository
                .GetTotalIncome()
                .GroupBy(i => new { i.Category, i.Company })
                .Select(g => new SaleItemAggregate {
                    Category = g.Key.Category,
                    Company = g.Key.Company,
                    Value = g.Sum(i => i.Income)
                })
                .ToList();
        }

        public MainViewModel() {
            this.repository = new SaleItemRepository();
        }
    }

    public class SaleItemAggregate {
        public string Company { get; set; }
        public string Category { get; set; }
        public double Value { get; set; }
    }
}

File: VB/PrintCrosshairSample/MainViewModel.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq

Namespace PrintCrosshairSample

    Public Class MainViewModel

        Private repository As SaleItemRepository

        Public ReadOnly Property TotalIncome As IReadOnlyList(Of SaleItemAggregate)
            Get
                Return repository.GetTotalIncome().GroupBy(Function(i) New With {i.Category, i.Company}).[Select](Function(g) New SaleItemAggregate With {g.Key.Category, g.Key.Company, .Value = g.Sum(Function(i) i.Income)}).ToList()
            End Get
        End Property

        Public Sub New()
            repository = New SaleItemRepository()
        End Sub
    End Class

    Public Class SaleItemAggregate

        Public Property Company As String

        Public Property Category As String

        Public Property Value As Double
    End Class
End Namespace
